src/context.ts

```diff
@@ -634,6 +634,45 @@ export default class Context {
 		return this._customFieldKeys();
 	}
 
+	private _customFieldValue(experimentName: string, key: string): JSONValue {
+		const experiment = this._index[experimentName];
+
+		if (experiment != null) {
+			const field = experiment.data.customFieldValues?.find((x) => x.name === key);
+			if (field != null) {
+				switch (field.type) {
+					case "text":
+					case "string":
+						return field.value;
+					case "number":
+						return Number(field.value);
+					case "json":
+						try {
+							return JSON.parse(field.value);
+						} catch (e) {
+							console.error(`Failed to parse JSON custom field value '${key}' for experiment '${experimentName}'`);
+							return null;
+						}
+					case "boolean":
+						return field.value === "true";
+					default:
+						console.error(
+							`Unknown custom field type '${field.type}' for experiment '${experimentName}' and key '${key}'`
```

Comment:
Here we could add in the message "You may need to upgrade SDK to latest version"